File: src/main/java/net/atomique/ksar/XML/GraphConfig.java
package net.atomique.ksar.XML;

import java.util.LinkedHashMap;

public class GraphConfig {

    public GraphConfig(String s1, String s2, String s3) {
        name = s1;
        Title = s2;
        type= s3;
        plotlist = new LinkedHashMap<>();
        stacklist = new LinkedHashMap<>();
    }
    public String getTitle() {
        return Title;
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }
    
    public void addPlot(PlotStackConfig s) {
        plotlist.put(s.getTitle(), s);
    }

    public LinkedHashMap<String, PlotStackConfig> getPlotlist() {
        return plotlist;
    }

    public void addStack(PlotStackConfig s) {
        stacklist.put(s.getTitle(), s);
    }

    public LinkedHashMap<String, PlotStackConfig> getStacklist() {
        return stacklist;
    }

    private String name =null;
    private String Title = null;
    private String type = null;
    private LinkedHashMap<String,PlotStackConfig> plotlist;
    private LinkedHashMap<String,PlotStackConfig> stacklist;
}

File: src/main/kotlin/net/atomique/ksar/xml/GraphConfig.kt
/*
 * Copyright 2018 The kSAR Project. All rights reserved.
 * See the LICENSE file in the project root for more information.
 */
package net.atomique.ksar.xml

class GraphConfig(val name: String, val title: String, val type: String) {
    val plotlist = mutableMapOf<String, PlotStackConfig>()
    val stacklist = mutableMapOf<String, PlotStackConfig>()

    fun addPlot(s: PlotStackConfig) {
        plotlist[s.title] = s
    }

    fun addStack(s: PlotStackConfig) {
        stacklist[s.title] = s
    }
}
